src/hotspot/share/opto/castnode.hpp

```diff
@@ -67,18 +142,20 @@ class ConstraintCastNode: public TypeNode {
   virtual Node *Ideal(PhaseGVN *phase, bool can_reshape);
   virtual int Opcode() const;
   virtual uint ideal_reg() const = 0;
-  virtual bool depends_only_on_test() const { return _dependency == RegularDependency; }
-  bool carry_dependency() const { return _dependency != RegularDependency; }
+  bool carry_dependency() const { return !_dependency.cmp(FloatingNarrowingDependency); }
+  virtual bool depends_only_on_test() const { return _dependency.floating(); }
```

Comment:
Why not rename it to `is_dependency_floating`? That may be more helpful at the use site.